csharp/Connection/TypeDBSession.cs

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+
+using TypeDB.Driver;
+using TypeDB.Driver.Api;
+using TypeDB.Driver.Common;
+
+namespace TypeDB.Driver.Connection
+{
+    public class TypeDBSession : NativeObjectWrapper<Pinvoke.Session>, ITypeDBSession
+    {
+        private readonly SessionType _type;
+        private readonly TypeDBOptions _options;
+
+        private readonly List<SessionCallback> _callbacks;
+
+        public TypeDBSession(IDatabaseManager databaseManager, string database, SessionType type, TypeDBOptions options)
```

Comment:
I just understood that it is too public, needs to be available only for our executable so (internal).  Gonna review all the constructors on Monday.